amount.go

```diff
@@ -148,6 +148,12 @@ func (a Amount) Convert(currencyCode, rate string) (Amount, error) {
 
 // Add adds a and b together and returns the result.
 func (a Amount) Add(b Amount) (Amount, error) {
+	if a.Equal(Amount{}) {
```

Comment:
This does cause a 25% performance regression in the benchmark. If you preferred, at the cost of some readability, I could move these checks to instide the `if a.currencyCode != b.currencyCode` block at line 152.

Reply:
Thank you, I think that was the right decision.